TestHelper/MemoryLogger.cs

```diff
@@ -0,0 +1,38 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+namespace Microsoft.Authentication.TestHelper
+{
+    using NLog;
+    using NLog.Config;
+    using NLog.Targets;
+
+    using System.Diagnostics;
+
+    public static class MemoryLogger
+    {
+        public static (Extensions.Logging.ILogger logger, MemoryTarget logTarget) Create()
+        {
+            // Use reflection to get the name of the class which declares our caller.

```

Comment:
Is using reflection a good idea here? Any design implications?

Reply:
Somewhat covered in the description - I chose reflection here after initially taking a generic paramter `T` to create `logger` as an `ILogger<T>` to set the name. But then I copy and pasted the wrong class arg into all the tests. Since no assertions use it yet, no tests failed, but in the future the name would not have been expected or made sense. So with reflection, you can copy and paste the same line and it does the right thing in each test automatically :)